From 3cdc76b7ec0e058a6c216521c6e9dc6914c79a23 Mon Sep 17 00:00:00 2001
From: Graham Inggs <graham.inggs@uct.ac.za>
Date: Tue, 12 Mar 2013 17:02:00 +0000
Subject: [PATCH 03/03] Link uil executable against libUil

The uil executable can be linked against libUil instead of compiling with the
libUil code.  This results in a reduction in size of the uil executable from
about 348KB to about 16KB on amd64.

This patch originates from Red Hat, but I am unable to find a changelog entry or
bug number and it does not appear to have been forwarded either.

Upstream-Status: Submitted [http://bugs.motifzone.net/show_bug.cgi?id=1586]
---
 clients/uil/Makefile.am |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/clients/uil/Makefile.am
+++ b/clients/uil/Makefile.am
@@ -13,7 +13,7 @@
 
 libUil_la_LIBADD = UilParser.lo ../../lib/Mrm/libMrm.la ../../lib/Xm/libXm.la 
 
-uil_LDADD = ../../lib/Mrm/libMrm.la ../../lib/Xm/libXm.la 
+uil_LDADD = libUil.la ../../lib/Mrm/libMrm.la ../../lib/Xm/libXm.la
 
 INCLUDES = -DINCDIR=\"@INCDIR@\" \
            -DLIBDIR=\"@LIBDIR@\" \
@@ -42,7 +42,7 @@
 		UilLstMac.c	UilSemVal.c	UilSemCSet.c	UilDB.c
 
 SRCS =		$(COMMON_SRC)
-SRCS2 =		$(COMMON_SRC)	UilMain.c
+SRCS2 =		UilMain.c
 
 HEADERS_1 = 	Uil.h		UilSymGl.h	UilSymDef.h \
 		UilDef.h	XmAppl.uil
